src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1639,19 +1637,17 @@ private TokenCacheItem findFamilyItemForUser(final UserIdentifierType userIdenti
             return null;
         }
 
-        Iterator<TokenCacheItem> allItems = mTokenCacheStore.getAll();
-        if (allItems == null || !allItems.hasNext()) {
+        Iterable<TokenCacheItem> allItems = mTokenCacheStore.getAll();
+        if (allItems == null || !allItems.iterator().hasNext()) {
```

Comment:
For your purposes I think that getAll should return actually Collection<> (Not just iterable)...
It has method isEmpty()